### PR TITLE
Automated cherry pick of #83196: using online instead to fix kubelet service failed with wrong

### PR DESCRIPTION
**Pull request by zouyee**
Cherry pick of #83196 on release-1.16.

#83196: using online instead to fix kubelet service failed with wrong

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Use online nodes instead of possible nodes when discovering available NUMA nodes
```

### CODE REVIEW

**Review by odinuge — APPROVED** [review]
/lgtm

**Comment by odinuge** [comment]
Just another note @zouyee, it would have been nice with a changelog entry for this change. This applies to both this PR and the original one.

Here is just a simple one, but feel free to use something else;

`````
```release-note
Use online nodes instead of possible nodes when discovering available NUMA nodes
```
`````

**Comment by dougm** [comment]
/kind bug
/kind flake

@zouyee can you also copy-n-paste the `release-note` to #83196 ?

**Comment by zouyee** [comment]
Done!

**Comment by zouyee** [comment]
@dougm PTAL

**Comment by odinuge** [comment]
This is just awaiting the `approval` label, and that has to be set by some of the sig-node approvers.

You can see the status here: https://prow.k8s.io/pr?query=is%3Apr+repo%3Akubernetes%2Fkubernetes+author%3Azouyee+head%3Aautomated-cherry-pick-of-%2383196-upstream-release-1.16

**Comment by derekwaynecarr** [comment]
this is a bug that prevents gathering valuable feedback as we look to promote topology manager further.

/approve
/lgtm

**Pull request merged**
